lib/Backend/Inline.cpp

```diff
@@ -4799,7 +4799,26 @@ Inline::MapFormals(Func *inlinee,
                     // "this" is a constant, so map it now.
                     // Don't bother mapping if it's not an object, though, since we'd have to create a
                     // boxed value at JIT time, and that case doesn't seem worth it.
-                    Js::TypeId typeId = Js::JavascriptOperators::GetTypeIdNoCheck(thisConstSym->GetConstAddress(topFunc->IsOOPJIT()));
+                    Js::TypeId typeId = Js::TypeIds_Limit;
+                    Js::Var localVar = thisConstSym->GetConstAddress(topFunc->IsOOPJIT());
+                    if (localVar != nullptr)
+                    {
+                        typeId = Js::JavascriptOperators::GetTypeIdNoCheck(localVar);
+                    }
+                    else
+                    {
+                        Assert(JITManager::GetJITManager()->IsJITServer());
+                        // with OOP JIT we may create const Opnds for library vars without materializing a JITRecyclableObject
+                        IR::Opnd * thisConstOpnd = thisConstSym->GetConstOpnd();
+                        if (thisConstOpnd->GetValueType().IsUndefined())
+                        {
+                            typeId = Js::TypeIds_Undefined;
+                        }
+                        else if (thisConstOpnd->GetValueType().IsNull())
+                        {
+                            typeId = Js::TypeIds_Null;
+                        }
```

Comment:
I thought there might be a way for it to be some other library value that we wouldn't optimize. Since condition below has checks that would exclude other library values, I thought it should be ok not to check for them.